scc/light_client/provider/rpc.go

```diff
@@ -0,0 +1,118 @@
+package provider
+
+import (
+	"fmt"
+
+	"github.com/0xsoniclabs/sonic/ethapi"
+	"github.com/0xsoniclabs/sonic/scc"
+	"github.com/0xsoniclabs/sonic/scc/cert"
+	"github.com/Fantom-foundation/lachesis-base/inter/idx"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// RPCProvider implements the Provider interface and provides methods
+// making RPC calls through an Ethereum client.
+type RPCProvider struct {
```

Comment:
nit: naming convention wise I would name this type `RpcProvider`. Event though `RPC` is an acronym, it makes it easier if we just start every word in the name with a capital letter ...